todoapp/TODOapp/app/src/main/java/com/nearsoft/androidschool/todoapp/activities/map/MapsActivity.java

```diff
@@ -37,9 +38,10 @@ protected void onCreate(Bundle savedInstanceState) {
     public void onMapReady(GoogleMap googleMap) {
         map = googleMap;
 
-        LatLng latLng = new LatLng(-34, 151);
-        map.addMarker(new MarkerOptions().position(latLng).title(toDoContent.getTitle()));
-        map.moveCamera(CameraUpdateFactory.newLatLng(latLng));
+//        TODO get toDoLatLng from toDoContent variable
```

Comment:
The whole point of the Map and Location feature is to be able to see where the TODO was taken.

You should always obtain the location of a new Todo.

In order to achieve this:

1. Modify the Todo Model to add the location, it could be double lat double lng, or something like that, or a Location model inside the Todo.
2. Add stub data (in order to test the new feature)
3. Program it.

You do not need "add" Todo feature in order to make this work, you need "add" todo to add the LocationManager or to save data, but to show the Map marker you do not need this.